File: packages/app/src/services/software/definitions/javaSdk.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'java-sdk';
	public name = 'Java SE Development Kit';
	public category = [SoftwareCategory.Development, SoftwareCategory.Runtime];
	public downloadName = '';
	public icon = 'java.png';
	public homepage = 'https://oracle.com/java/technologies/downloads';

	public async resolveDownloadUrl() {
		return ok('');
	}
}

File: src/lib/software/java_sdk.py
from src.lib.software import BaseSoftware, SoftwareCategory
from src.lib.software.variants.java_sdk_21 import JavaSEDevelopmentKit21
from src.lib.software.variants.java_sdk_25 import JavaSEDevelopmentKit25

class JavaSEDevelopmentKit(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.key = 'java-sdk'
        self.name = 'Java SE Development Kit'
        self.category = [SoftwareCategory.Development, SoftwareCategory.Runtime]
        self.variants = [JavaSEDevelopmentKit25(), JavaSEDevelopmentKit21()]
        self.icon = 'java.png'
        self.homepage = 'https://oracle.com/java/technologies/downloads'

    def resolve_download_url(self):
        pass
